typing/printpat.ml

```diff
@@ -94,8 +94,12 @@ let rec pretty_val : type k . _ -> k general_pattern -> _ = fun ppf v ->
           fprintf ppf "@[{%a%t}@]"
             pretty_lvals filtered_lvs elision_mark
       end
-  | Tpat_array vs ->
-      fprintf ppf "@[[| %a |]@]" (pretty_vals " ;") vs
+  | Tpat_array (am, vs) ->
+      let punct = match am with
+        | Mutable   -> '|'
+        | Immutable -> ':'
+      in
+      fprintf ppf "@[[%c %a %c]@]" punct (pretty_vals " ;") vs punct
```

Comment:
This is using the former concrete syntax for immutable arrays, which I think we haven't kept for now. I would use just `[| ... |]` (betting on type-based disambiguation) or maybe a fake attribute.